Sources/System/MachPort.swift

```diff
@@ -0,0 +1,318 @@
+                let secret = mach_port_context_t(arc4random())
+                let kr = mach_port_guard(mach_task_self_, name, secret, 0)
+                assert(kr == KERN_SUCCESS)
+                self.context = secret
+            }
+            else {
+                self.context = 0
+            }
+        }
+
+        /// Borrow access to the port name in a block that can perform
+        /// non-consuming operations.
+        ///
+        /// Take care when using this function; many operations consume rights,
+        /// and send-once rights are easily consumed.
+        ///
+        /// If the right is consumed, behavior is undefined.
+        ///
+        /// The body block may optionally return something, which will then be
+        /// returned to the caller of withBorrowedName.
+        func withBorrowedName<ReturnType>(body:(mach_port_name_t) -> ReturnType) -> ReturnType {
+            return body(name)
+        }
+
+        deinit {
+            if name != 0xFFFFFFFF /* MACH_PORT_DEAD */ {
+                if RightType.self == ReceiveRight.self {
+                    // recv rights must be mod ref'ed instead of deallocated
+                    let kr = mach_port_unguard(mach_task_self_, name, context)
+                    assert(kr == KERN_SUCCESS)
+
+                    let kr2 = mach_port_mod_refs(mach_task_self_, name, MACH_PORT_RIGHT_RECEIVE, -1)
+                    assert(kr2 == KERN_SUCCESS)
+                } else {
+                    mach_port_deallocate(mach_task_self_, name)
+                }
+            }
+        }
+    }
+
+    /// Possible errors that can be thrown by Mach.Port operations.
+    enum PortRightError : Error {
+        /// Returned when an operation cannot be completed, because the Mach
+        /// port right has become a dead name. This is caused by deallocation of the
+        /// receive right on the other end.
+        case deadName
+    }
+
+    /// The MachPortRight type used to manage a receive right.
+    struct ReceiveRight : MachPortRight {}
+
+    /// The MachPortRight type used to manage a send right.
+    struct SendRight : MachPortRight {}
+
+    /// The MachPortRight type used to manage a send-once right.
+    ///
+    /// Send-once rights are the most restrictive type of Mach port rights.
+    /// They cannot create other rights, and are consumed upon use.
+    ///
+    /// Upon destruction a send-once notification will be sent to the
+    /// receiving end.
+    struct SendOnceRight : MachPortRight {}
+
+    /// Create a connected pair of rights, one receive, and one send.
+    ///
+    /// This function will abort if the rights could not be created.
+    /// Callers may assert that valid rights are always returned.
+    static func allocatePortRightPair() -> (Mach.Port<Mach.ReceiveRight>, Mach.Port<Mach.SendRight>) {
+        var name = mach_port_name_t(MACH_PORT_NULL)
+        let secret = mach_port_context_t(arc4random())
+        withUnsafeMutablePointer(to: &name) { name in
```

Comment:
Remove the `var name` and replace with something that lets you use `let`:

```suggestion
        let name = withUnsafeMutablePointer(to: &name) { name in
             var options = mach_port_options_t()
             options.flags = UInt32(MPO_INSERT_SEND_RIGHT);
             return withUnsafeMutablePointer(to: &options) { options in
                 let kr = mach_port_construct(mach_task_self_, options, secret, name)
                 assert(kr == KERN_SUCCESS)
                 return name
             }
         }
```

Reply:
The compiler doesn't let me use it for the `to:` argument in the same statement that declares it.